### ISSUE TITLE
[Bug]: Transactions on Arbitrum with the nonce+1 consistently fail, resulting in an 'Internal JSON-RPC' error message

### ISSUE THREAD

**Issue by sleepytanya** [MetaMask/metamask-extension#27638]
### Describe the bug

Transactions on Arbitrum with the nonce+1 consistently fail, resulting in an `Internal JSON-RPC error` message. 
The issue is present in versions 12.4.0 and 12.3.0, and likely affect earlier versions.

### Expected behavior

-

### Screenshots/Recordings

https://github.com/user-attachments/assets/0b36755b-4921-4b55-92d6-4d7b88fd6a0f

### Steps to reproduce

1. Connect to Arbitrum
2. Enable `Customize transaction nonce` in `Settings -> Advanced`
3. Submit this transaction

### Detection stage

During release testing

### Version

12.4.0, 12.3.0

### Build type

None

### Browser

Chrome, Firefox

### Operating system

MacOS

**Comment by vinistevam** [comment]
I encountered a consistent Nonce too high error while testing on Infura's Arbitrum RPC. However, when switching to a public RPC node, I was able to successfully send higher nonce transactions. These transactions were queued, and after sending transactions up to the queued nonce, all of them were confirmed as expected. Although I couldn't find any documentation from Infura explicitly stating that their Arbitrum RPC does not support transaction queuing, it appears that this functionality might be limited or unsupported on their RPC.

https://github.com/user-attachments/assets/b4766184-9845-439d-98a1-b57b9c924e5b
cc @sleepytanya
